tech-specs/src/explore/design/dkg.md

```diff
@@ -1 +1,154 @@
-# Distributed key generation gossip
+# Distributed key generation
+
+Here we give an overview of how the DKG protocol works with the ledger and 
+the roles of each node in the network per this protocol. This portion of the 
+book focuses solely on the production of a new public encryption key via the
+DKG scheme. It does not cover encryption, decryption, or validation of transactions.
+
+Some parts of the DKG protocol may be done through the gossip layer or through
+off-chain communication. However, the entire DKG cannot exist solely in the 
+gossip layer as consensus on the final shared secrets must be achieved. This 
+will require posting information on chain.
+
+## Starting the Protocol
+
+To initializing the protocol to generate a new key, the  current block 
+proposer proposes a new instance. This would be posted on the blockchain and
+notify all the participating validators to begin the announcement phase. 
+However, when should the proposer make this decision? In all likelihood it will be a combination of blocks transpired since
+the last instance completed as well as special cases (such as an instance
+failing to complete).
+
+## Announcing and distributing weight shares
+
+In order for the DKG protocol to work, it requires the contribution of 
+ \\(\frac{2}{3} \\) of the total staked weight among live validators. In 
+order to know when a subset of validators own at least \\(\frac{2}{3}\\)
+of the total weight, it is necessary for validators to announce their stake.
+
+Furthermore, the total weight will be scaled down to a fixed parameter 
+\\( W\\) (and fractional weights rounded appropriately to acheive integral
+values). After this procedure, each validator has a corresponding number of
+weight shares. 
+
+Lastly, the DKG protocol requires a canonical ordering of validators with
+shares must be agreed upon. This will be done via the consensus layer.
+The current block proposer will apportion out weight shares as well as the
```

Comment:
I think the partitioning should also happen implicitly. Assuming that, when the DKG start block height is reached, everyone agrees on the staking weights of the validator set, then everyone can locally compute the same partitioning - it doesn't have to be posted on the chain.

Reply:
We just need to specify a canonical ordering. Descending in terms of stake almost gets us there but we need a tie break on equally staked weights. Maybe validator addresses sorted lexicographically?